recipes/modsecurity.rb

```diff
@@ -44,17 +44,33 @@
   end
 end
 
-execute 'make_modsecurity' do
-  command <<-EOS
-  make clean
-  sh build.sh
-  git submodule init
-  git submodule update
-  ./configure
-  make
-  make install
-  EOS
-  environment 'CFLAGS' => "-DDEFAULT_USER=\\\"#{node['nginxxx']['user']}\\\" -DDEFAULT_GROUP=\\\"#{node['nginxxx']['user']}\\\""
-  cwd "/usr/local/src/modsecurity"
-  not_if 'test -d /usr/local/modsecurity'
+if node['nginxxx']['modsecurity']['force_build']
+  execute 'make_modsecurity' do
+    command <<-EOS
+    make clean
+    sh build.sh
+    git submodule init
+    git submodule update
+    ./configure
+    make
+    make install
+    EOS
+    environment 'CFLAGS' => "-DDEFAULT_USER=\\\"#{node['nginxxx']['user']}\\\" -DDEFAULT_GROUP=\\\"#{node['nginxxx']['user']}\\\""
+    cwd "/usr/local/src/modsecurity"
+  end
+else
+  execute 'make_modsecurity' do
+    not_if 'test -d /usr/local/modsecurity'
```

Comment:
How about detecting installation by `.so` file's version?
e.g. `not_if "/usr/local/modsecurity/lib/libmodsecurity.so.#{node['nginxxx']['modsecurity']['version']}"` ?

Reply:
Thanks a good comment.:pray:
I'll fix it.

Reply:
A-...but if this fix commit, rollback to old version after force build.:sob:



install ModSecurity version | 3.0.3 | master
-- | -- | --
branch | v3/master | v3/master
version | 3.0.3 | 3.0.3
force_build | false | true

Reply:
I think that to add `unless node['nginxxx']['modsecurity']['force_build']` at the end of this line better.

Reply:
Exactly so! :pray: